src/main/Config.h

```diff
@@ -634,5 +634,8 @@ class Config : public std::enable_shared_from_this<Config>
     void processOpApplySleepTimeForTestingConfigs();
 
     std::chrono::seconds HISTOGRAM_WINDOW_SIZE;
+
+    // How many times to retry fetching a history archive.
+    uint16_t HISTORY_ARCHIVE_RETRY_LIMIT;
```

Comment:
This should be a `uint32_t`. I'd recommend compiling with `./configure --enable-extrachecks` during development to get warnings for issues like type mismatch. This would have given you a cast error in `src/main/Config.cpp:1442` and may have helped you spot this bug.